ext/soap/soap.c

```diff
@@ -579,13 +580,15 @@ PHP_METHOD(SoapHeader, __construct)
 		if (ZSTR_LEN(actor_str) > 2) {
 			add_property_stringl(this_ptr, "actor", ZSTR_VAL(actor_str), ZSTR_LEN(actor_str));
 		} else {
-			php_error_docref(NULL, E_WARNING, "Invalid actor");
+			zend_argument_value_error(2, "must be longer than 2 characters");
+			RETURN_THROWS();
 		}
 	} else if (!actor_is_null) {
 		if ((actor_long == SOAP_ACTOR_NEXT || actor_long == SOAP_ACTOR_NONE || actor_long == SOAP_ACTOR_UNLIMATERECEIVER)) {
 			add_property_long(this_ptr, "actor", actor_long);
 		} else {
-			php_error_docref(NULL, E_WARNING, "Invalid actor");
+			zend_argument_value_error(5, "must be either SOAP_ACTOR_NEXT, SOAP_ACTOR_NONE or SOAP_ACTOR_UNLIMATERECEIVER");
```

Comment:
I have to wonder wtf UNLIMATERECEIVER is supposed to mean...

Reply:
:D Do you also suppose it's "ultimate receiver"? As far as I see, Google shows quite some hits for "soap ultimate receiver" :D